stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -907,15 +910,12 @@ internal constructor(
                 false -> Unit
             }
         } else {
-            when (chatSocketExperimental.isDisconnected()) {
```

Comment:
Connection State shouldn't be checked outside of the ChatSocket implementation.
It should be handled by the internal StateMachine that know if the reconnection need to be done or not